beacon/engine/types.go

```diff
@@ -265,15 +265,7 @@ func ExecutableDataToBlockNoHash(data ExecutableData, versionedHashes []common.H
 
 	var requestsHash *common.Hash
 	if requests != nil {
-		// Put back request type byte.
-		typedRequests := make([][]byte, len(requests))
-		for i, reqdata := range requests {
-			typedReqdata := make([]byte, len(reqdata)+1)
-			typedReqdata[0] = byte(i)
-			copy(typedReqdata[1:], reqdata)
-			typedRequests[i] = typedReqdata
-		}
-		h := types.CalcRequestsHash(typedRequests)
+		h := types.CalcRequestsHash(requests)
```

Comment:
I am a bit confused here. 

The requests is supplied by consensus client and originally the requests have no prefix type byte.

Do we change the API spec that the requests must have the prefix type byte now?

Reply:
Yes it's the change I am making to the engine API spec. Link in description.